src/loaddb/load_db_value_converter.cpp

```diff
@@ -946,4 +947,67 @@ namespace cubload
     assert (0);
     return ER_FAILED;
   }
+
+  int
+  to_int_generic (const char *str, const attribute *attr, db_value *val)
+  {
+    size_t str_len = strlen (str);
+    int error_code = NO_ERROR;
+
+    db_make_int (val, 0);
+
+    /* Let try take the fastest path here, if we know that number we are getting fits into a long, use strtol, else we
+     * need to convert it to a double and coerce it, checking for overflow. Note if integers with leading zeros are
+     * entered this can take the slower route. */
+    if (str_len < MAX_DIGITS_FOR_INT || (str_len == MAX_DIGITS_FOR_INT && (str[0] == '0' || str[0] == '1')))
+      {
+	error_code = parse_int (&val->data.i, str, 10);
+	if (error_code != NO_ERROR)
+	  {
+	    er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IT_DATA_OVERFLOW, 1, pr_type_name (DB_TYPE_INTEGER));
+	    return ER_IT_DATA_OVERFLOW;
+	  }
+      }
+    else if (str_len < MAX_DIGITS_FOR_BIGINT || (str_len == MAX_DIGITS_FOR_BIGINT && str[0] != '9'))
+      {
+	db_make_bigint (val, 0);
+	error_code = parse_bigint (&val->data.bigint, str, 10);
+	if (error_code != NO_ERROR)
+	  {
+	    er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IT_DATA_OVERFLOW, 1, pr_type_name (DB_TYPE_BIGINT));
+	    return ER_IT_DATA_OVERFLOW;
+	  }
+      }
+    else
+      {
+	DB_NUMERIC num;
+	DB_BIGINT tmp_bigint;
+
+	numeric_coerce_dec_str_to_num (str, num.d.buf);
+	if (numeric_coerce_num_to_bigint (num.d.buf, 0, &tmp_bigint) != NO_ERROR)
+	  {
+	    error_code = db_value_domain_init (val, DB_TYPE_NUMERIC, str_len, 0);
+	    if (error_code != NO_ERROR)
+	      {
+		er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IT_DATA_OVERFLOW, 1, pr_type_name (DB_TYPE_BIGINT));
```

Comment:
why is this an overflow error?

Reply:
the only case that the function returns an error is str_len > max precision of numeric.
I'm not sure this holds.